pkg/kms/service.go

```diff
@@ -179,6 +184,63 @@ func (s *service) VerifyMAC(keystoreID, keyID string, mac, data []byte) error {
 	return nil
 }
 
+// WrapKey wraps cek for the recipient with public key 'recipientPubKey'.
+func (s *service) WrapKey(keystoreID, keyID string, cek, apu, apv []byte,
+	recipientPubKey *crypto.PublicKey) (*crypto.RecipientWrappedKey, error) {
+	if keyID != "" {
+		kh, err := s.getKeyHandle(keystoreID, keyID)
+		if err != nil {
+			return nil, err
+		}
+
+		// ECDH-1PU key wrapping (Authcrypt)
+		recipientWrappedKey, err := s.crypto.WrapKey(cek, apu, apv, recipientPubKey, crypto.WithSender(kh))
+		if err != nil {
+			return nil, NewServiceError(wrapKeyFailed, err)
+		}
+
+		return recipientWrappedKey, nil
+	}
+
+	recipientWrappedKey, err := s.crypto.WrapKey(cek, apu, apv, recipientPubKey)
+	if err != nil {
+		return nil, NewServiceError(wrapKeyFailed, err)
+	}
+
+	return recipientWrappedKey, nil
+}
+
+// UnwrapKey unwraps a key in recipientWK.
+func (s *service) UnwrapKey(keystoreID, keyID string, recipientWK *crypto.RecipientWrappedKey,
+	senderPubKey *crypto.PublicKey) ([]byte, error) {
+	kh, err := s.getKeyHandle(keystoreID, keyID)
+	if err != nil {
+		return nil, err
+	}
+
+	if senderPubKey != nil {
+		senderKH, e := keyio.PublicKeyToKeysetHandle(senderPubKey)
+		if e != nil {
+			return nil, NewServiceError(unwrapKeyFailed, e)
+		}
```

Comment:
this conversation is unnecessary

`crypto.WithSender()`
now accepts `*crypto.PublicKey` directly

Reply:
Thanks @Baha-sk, I will update this in my next PR.